src/diffusers/pipelines/unidiffuser/pipeline_unidiffuser.py

```diff
@@ -158,6 +135,35 @@ def __init__(
         # TODO: handle safety checking?
         self.safety_checker = None
 
+    def enable_vae_slicing(self):
+        r"""
+        Enable sliced VAE decoding. When this option is enabled, the VAE will split the input tensor in slices to
+        compute decoding in several steps. This is useful to save some memory and allow larger batch sizes.
+        """
+        self.vae.enable_slicing()
+
+    def disable_vae_slicing(self):
+        r"""
+        Disable sliced VAE decoding. If `enable_vae_slicing` was previously enabled, this method will go back to
+        computing decoding in one step.
+        """
+        self.vae.disable_slicing()
+
+    def enable_vae_tiling(self):
+        r"""
+        Enable tiled VAE decoding. When this option is enabled, the VAE will split the input tensor into tiles to
+        compute decoding and encoding in several steps. This is useful for saving a large amount of memory and to allow
+        processing larger images.
+        """
+        self.vae.enable_tiling()
+
+    def disable_vae_tiling(self):
+        r"""
+        Disable tiled VAE decoding. If `enable_vae_tiling` was previously enabled, this method will go back to
+        computing decoding in one step.
+        """
+        self.vae.disable_tiling()
+
     # Modified from diffusers.pipelines.stable_diffusion.pipeline_stable_diffusion.StableDiffusionPipeline.enable_model_cpu_offload
     # Add self.image_encoder, self.text_decoder to cpu_offloaded_models list
     def enable_model_cpu_offload(self, gpu_id=0):
```

Comment:
Do we still need to add it as it's now a part of `DiffusionPipeline`.

Reply:
I guess I'm a little confused about the API; `DiffusionPipeline` has a [`enable_sequential_cpu_offload`](https://huggingface.co/docs/diffusers/v0.20.0/en/api/diffusion_pipeline#diffusers.DiffusionPipeline.enable_sequential_cpu_offload) method, but its docstring says that it has higher memory savings and lower performance than `enable_model_cpu_offload`, which is not currently a method of `DiffusionPipeline`. This makes it seem like `enable_sequential_cpu_offload` and `enable_model_cpu_offload` are two different memory-saving methods with different tradeoffs, but `enable_model_cpu_offload` has to be implemented in the child classes of `DiffusionPipeline`.

Comparing the implementation of [`DiffusionPipeline.enable_sequential_cpu_offload`](https://github.com/huggingface/diffusers/blob/v0.20.0/src/diffusers/pipelines/pipeline_utils.py#L1202) and e.g. [`StableDiffusionPipeline.enable_model_cpu_offload`](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L225) (on which `UniDiffuser.enable_model_cpu_offload` is based), the two methods are pretty similar except for the fact that the former uses [`accelerate.cpu_offload`](https://huggingface.co/docs/accelerate/v0.22.0/en/package_reference/big_modeling#accelerate.cpu_offload) and the latter uses [`accelerate.cpu_offload_with_hook`](https://github.com/huggingface/accelerate/blob/main/src/accelerate/big_modeling.py#L207). Both functions seem like sensible strategies for reducing the memory usage of diffusion pipeline inference since they typically run the `unet` (and possibly other models) `forward` in each iteration of the sampling loop.

Is `DiffusionPipeline.enable_sequential_cpu_offload` intended to supplant usage of any `enable_model_cpu_offload` methods?

Reply:
> but its docstring says that it has higher memory savings and lower performance than enable_model_cpu_offload, which is not currently a method of DiffusionPipeline.

For extremely low GPU VRAM instances, `enable_sequential_cpu_offload()` should be preferred. @patrickvonplaten is working on making `enable_model_cpu_offload()` a part of `DiffusionPipeline` class. 

For additional references, I welcome you to check out the `accelerate` docs here: https://huggingface.co/docs/accelerate/package_reference/big_modeling. 

Also ccing @muellerzr in case he has to offer any additional thoughts / explanations regarding sequential CPU offload and model CPU offload.

Reply:
Saw that there is an open PR #4514 that refactors the `enable_model_cpu_offload` method to be a method of `DiffusionPipeline`. Is it better to wait until that PR is merged and then merge the changes into this PR or to preemptively remove `UniDiffuser.enable_model_cpu_offload` here?

Reply:
Sure we can wait :)

Reply:
Merged in commit 9357965 to the PR branch and removed `UniDiffuserPipeline.enable_model_cpu_offload`.